frontend/package.json

```
@@ -17,16 +19,30 @@
     "vite-tsconfig-paths": "4.3.2"
   },
   "devDependencies": {
+    "@babel/preset-env": "7.24.3",
+    "@babel/preset-react": "7.24.1",
+    "@jest/globals": "29.7.0",
+    "@testing-library/jest-dom": "6.4.2",
+    "@testing-library/react": "14.2.2",
+    "@testing-library/user-event": "14.5.2",
+    "@types/jest": "29.5.12",
     "@types/node": "20.11.30",
     "@types/react": "18.2.71",
     "@types/react-dom": "18.2.22",
     "@typescript-eslint/eslint-plugin": "7.4.0",
     "@typescript-eslint/parser": "7.4.0",
+    "babel-jest": "29.7.0",
```

```suggestion
    "@jest/globals": "29.7.0",
    "@testing-library/jest-dom": "6.4.2",
    "@testing-library/react": "14.2.2",
    "@testing-library/user-event": "14.5.2",
    "@types/jest": "29.5.12",
    "@types/node": "20.11.30",
    "@types/react": "18.2.71",
    "@types/react-dom": "18.2.22",
    "@typescript-eslint/eslint-plugin": "7.4.0",
    "@typescript-eslint/parser": "7.4.0",
```
Remove as they are not used